src/main/java/org/zalando/failsafeactuator/CircuitBreakersEndpoint.java

```diff
@@ -0,0 +1,87 @@
+package org.zalando.failsafeactuator;
+
+import net.jodah.failsafe.CircuitBreaker;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.annotation.Selector;
+import org.springframework.boot.actuate.endpoint.annotation.WriteOperation;
+import org.springframework.context.ApplicationContext;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import static java.util.stream.Collectors.toMap;
+
+@Endpoint(id = "circuit-breakers")
+public class CircuitBreakersEndpoint {
+
+    private final ApplicationContext context;
+
+    CircuitBreakersEndpoint(final ApplicationContext context) {
+        this.context = context;
+    }
+
+    @ReadOperation
+    public Map<String, CircuitBreakerView> circuitBreakers() {
+        return getCircuitBreakers().entrySet().stream()
+                .collect(toMap(Map.Entry::getKey, entry -> toView(entry.getValue())));
+    }
+
+    @ReadOperation
+    @Nullable
+    public CircuitBreakerView circuitBreaker(@Selector final String name) {
+        final CircuitBreaker breaker = getCircuitBreaker(name);
+
+        if (breaker == null) {
+            return null;
+        }
+
+        return toView(breaker);
+    }
+
+    @WriteOperation
+    @Nullable
+    public CircuitBreakerView transitionTo(@Selector final String name, final CircuitBreaker.State state) {
+        final CircuitBreaker breaker = getCircuitBreaker(name);
+
+        if (breaker == null) {
+            return null;
+        }
+
+        transitioner(state).accept(breaker);
+
+        return toView(breaker);
+    }
+
+    private Consumer<CircuitBreaker> transitioner(final CircuitBreaker.State state) {
+        switch (state) {
+            case OPEN:
+                return CircuitBreaker::open;
+            case HALF_OPEN:
+                return CircuitBreaker::halfOpen;
+            default:
```

Comment:
Where is `case CLOSED`? :p IMO this is confusing - can you test a potential `default` branch with a `null` state?

Reply:
Switch statements don't support null.

Reply:
There is no way to cover a default branch for an enum that is fully covered with individual cases. I usually either do this trick here or use an immutable map instead.

Reply:
I'd rather leave the potential `default` branch untested and have an explicit `case CLOSED` - I think the clarity wins over branch coverage here